presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/TransformCorrelatedSingleRowSubqueryToProject.java

```diff
@@ -72,13 +72,28 @@ public Result apply(LateralJoinNode parent, Captures captures, Context context)
                 .where(ProjectNode.class::isInstance)
```

Comment:
Use:
```
where(node -> node instanceof ProjectNode && !node.getOutputSymbols().equals(parent.getCorrelation()))
```

Then please remove all the changes below.

Reply:
Thanks, this looks much cleaner

Reply:
Not only cleaner, it has different semantic due the usage of `parent.getCorrelation()`

Reply:
Can you help me understand what is the difference between this and the earlier implementation? If I understood correctly, LateralJoin's correlation will always be a subset of its input's output symbols, and in the previous implementation, we were removing the projections whose output symbols are completely contained in parent's input's symbols. I understand that we are ignoring such projections during search itself, rather than removing them afterwards, but is there any other semantic difference, please?

Reply:
That is all.